diff --git a/dist/FormReact.d.ts b/dist/FormReact.d.ts
index 2db278fbbda6d9d0d3aa9ab49353360c8d7d99e4..9533f6e594eb1ba50635ba0830ecbd4eef6e94d0 100644
--- a/dist/FormReact.d.ts
+++ b/dist/FormReact.d.ts
@@ -1,5 +1,5 @@
-import * as Atom from "@effect-atom/atom/Atom";
-import type * as Registry from "@effect-atom/atom/Registry";
+import * as Atom from "effect/unstable/reactivity/Atom";
+import type * as Registry from "effect/unstable/reactivity/AtomRegistry";
 import { Field, FormAtoms } from "@lucas-barake/effect-form";
 import type { FieldState as FieldStateModule, Mode } from "@lucas-barake/effect-form";
 import type * as FormBuilder from "@lucas-barake/effect-form/FormBuilder";
diff --git a/dist/FormReact.js b/dist/FormReact.js
index c6551bcca928a167369d898e5a71d8d0fc5b737c..e2fc5f33151b573c881fd31d519f2e07f4178fd3 100644
--- a/dist/FormReact.js
+++ b/dist/FormReact.js
@@ -1,6 +1,6 @@
 import { jsx as _jsx, Fragment as _Fragment } from "react/jsx-runtime";
-import { RegistryContext, useAtom, useAtomMount, useAtomSet, useAtomValue } from "@effect-atom/atom-react";
-import * as Atom from "@effect-atom/atom/Atom";
+import { RegistryContext, useAtom, useAtomMount, useAtomSet, useAtomValue } from "@effect/atom-react";
+import * as Atom from "effect/unstable/reactivity/Atom";
 import { Field, FormAtoms } from "@lucas-barake/effect-form";
 import { getNestedValue } from "@lucas-barake/effect-form/Path";
 import * as Layer from "effect/Layer";
diff --git a/package.json b/package.json
index 34f7b2fb24ea0ac2a16a5d222ab972e310da687a..dbba9c1863a38361c9748d4959d53a39f0f2b543 100644
--- a/package.json
+++ b/package.json
@@ -44,9 +44,8 @@
     "@vitejs/plugin-react": "^5.1.1"
   },
   "peerDependencies": {
-    "@effect-atom/atom": "^0.5.0",
-    "@effect-atom/atom-react": "^0.5.0",
-    "effect": "^3.19.15",
+    "@effect/atom-react": "^4.0.0-beta.65",
+    "effect": "^4.0.0-beta.65",
     "react": "^18.0.0 || ^19.0.0"
   },
   "scripts": {
